# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [25]:
## Code goes here
import csv
Firstnames= []
Lastnames=[]
Dirty_Birthmonths=[]
Birthdays=[]
Dirty_Birthyears=[]
Dirty_Courses=[]
Dirty_Phonenumbers=[]
with open('recweekdata.txt') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            print(f'Column names are {", ".join(row)}')
            line_count += 1
        else:
            Firstnames.append(row[0])
            Lastnames.append(row[1])
            Dirty_Birthmonths.append(row[2])
            Birthdays.append(row[3])
            Dirty_Birthyears.append(row[4])
            Dirty_Courses.append(row[5])
            Dirty_Phonenumbers.append(row[6])
            line_count += 1
            
    print(f'Processed {line_count} lines.')
    print(Birthmonths)

Column names are firstname, lastname, birthmonth, birthday, birthyear, course, phonenumber
Processed 3001 lines.
['11', 'May', '5', 'Dec', 'Sep', 'March', '11', 'Aug', 'July', '9', 'December', '5', 'January', '1', '6', '4', '1', 'Apr', '12', '3', '7', '9', '4', 'Oct', '5', '12', 'March', '11', '10', 'May', '4', 'Jul', '12', 'Oct', '7', 'December', '2', '8', 'May', '2', 'Nov', 'September', 'Jan', 'February', 'January', 'Dec', 'August', 'Apr', '10', 'March', '8', '3', 'Dec', '7', 'May', '9', 'May', 'April', 'April', 'August', '9', 'Dec', 'Feb', 'November', 'Mar', 'September', 'Dec', 'November', 'November', 'May', '8', 'December', 'Jun', 'Oct', 'Sep', '2', 'May', '9', 'December', '8', 'Feb', '11', 'Jun', 'February', 'Jan', 'Nov', 'August', 'January', '3', 'Mar', 'Aug', 'Dec', 'Jun', 'November', '9', 'Mar', 'Jan', 'Jan', 'May', '5', 'April', 'August', 'Dec', 'July', 'Mar', '8', 'April', 'May', '8', 'January', '12', 'May', 'May', 'August', 'March', 'Jul', '1', 'Feb', 'Dec', '2', '3', 'Ma

In [29]:
#Birthmonth cleaning
#used technique in Day 14 discussions
Unique_Dirty_Birthmonths=set(Dirty_Birthmonths)
Birthmonth_Dict = {name:None for name in Unique_Dirty_Birthmonths}

for i in Birthmonth_Dict:
    cleaned_name = input(i+": "+"Enter name: ")
    Birthmonth_Dict[i]=cleaned_name
Birthmonth_Dict

June: Enter name: 6
May: Enter name: 5
8: Enter name: 8
2: Enter name: 2
11: Enter name: 11
9: Enter name: 9
7: Enter name: 7
1: Enter name: 1
July: Enter name: 7
Mar: Enter name: 3
November: Enter name: 11
September: Enter name: 9
April: Enter name: 4
January: Enter name: 1
Jul: Enter name: 7
Apr: Enter name: 4
March: Enter name: 3
Oct: Enter name: 10
12: Enter name: 12
August: Enter name: 8
Sep: Enter name: 9
Feb: Enter name: 2
Dec: Enter name: 12
4: Enter name: 4
February: Enter name: 2
December: Enter name: 12
10: Enter name: 10
Jan: Enter name: 1
6: Enter name: 6
Aug: Enter name: 8
Jun: Enter name: 6
Nov: Enter name: 11
5: Enter name: 5
3: Enter name: 3


{'June': '6',
 'May': '5',
 '8': '8',
 '2': '2',
 '11': '11',
 '9': '9',
 '7': '7',
 '1': '1',
 'July': '7',
 'Mar': '3',
 'November': '11',
 'September': '9',
 'April': '4',
 'January': '1',
 'Jul': '7',
 'Apr': '4',
 'March': '3',
 'Oct': '10',
 '12': '12',
 'August': '8',
 'Sep': '9',
 'Feb': '2',
 'Dec': '12',
 '4': '4',
 'February': '2',
 'December': '12',
 '10': '10',
 'Jan': '1',
 '6': '6',
 'Aug': '8',
 'Jun': '6',
 'Nov': '11',
 '5': '5',
 '3': '3'}

In [34]:
import re
Cleaned_Birthmonths = [re.sub(r''.join(i),r''.join(Birthmonth_Dict[i]),i) for i in Dirty_Birthmonths]
Cleaned_Birthmonths

['11',
 '5',
 '5',
 '12',
 '9',
 '3',
 '11',
 '8',
 '7',
 '9',
 '12',
 '5',
 '1',
 '1',
 '6',
 '4',
 '1',
 '4',
 '12',
 '3',
 '7',
 '9',
 '4',
 '10',
 '5',
 '12',
 '3',
 '11',
 '10',
 '5',
 '4',
 '7',
 '12',
 '10',
 '7',
 '12',
 '2',
 '8',
 '5',
 '2',
 '11',
 '9',
 '1',
 '2',
 '1',
 '12',
 '8',
 '4',
 '10',
 '3',
 '8',
 '3',
 '12',
 '7',
 '5',
 '9',
 '5',
 '4',
 '4',
 '8',
 '9',
 '12',
 '2',
 '11',
 '3',
 '9',
 '12',
 '11',
 '11',
 '5',
 '8',
 '12',
 '6',
 '10',
 '9',
 '2',
 '5',
 '9',
 '12',
 '8',
 '2',
 '11',
 '6',
 '2',
 '1',
 '11',
 '8',
 '1',
 '3',
 '3',
 '8',
 '12',
 '6',
 '11',
 '9',
 '3',
 '1',
 '1',
 '5',
 '5',
 '4',
 '8',
 '12',
 '7',
 '3',
 '8',
 '4',
 '5',
 '8',
 '1',
 '12',
 '5',
 '5',
 '8',
 '3',
 '7',
 '1',
 '2',
 '12',
 '2',
 '3',
 '3',
 '4',
 '2',
 '11',
 '6',
 '3',
 '6',
 '10',
 '7',
 '2',
 '9',
 '4',
 '5',
 '10',
 '10',
 '6',
 '2',
 '4',
 '9',
 '6',
 '12',
 '11',
 '10',
 '2',
 '9',
 '12',
 '11',
 '9',
 '8',
 '5',
 '3',
 '11',
 '8',
 '12',
 '4',
 '8',
 '1',
 '10',
 '1

In [33]:
#Cleaning birthyears
Cleaned_Birthyears=[]
#all you have to do is add the first 2 digits or remove them to clean so I'll just settle for adding because I think it looks better and avoids confusion
for i in Dirty_Birthyears:
    if len(i)==4:
        Cleaned_Birthyears.append(i)
    else:
        if i[0] == 0:
            Cleaned_Birthyears.append('20'+i)
        else:
            Cleaned_Birthyears.append('19'+i)
Cleaned_Birthyears

['1998',
 '2001',
 '1997',
 '1997',
 '1997',
 '1999',
 '1998',
 '1999',
 '1997',
 '2001',
 '1998',
 '1997',
 '1999',
 '1999',
 '1998',
 '1999',
 '1997',
 '1997',
 '1997',
 '1997',
 '1999',
 '1999',
 '1997',
 '1997',
 '1997',
 '1999',
 '1997',
 '2000',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '2001',
 '1998',
 '2000',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '1999',
 '1998',
 '1997',
 '1998',
 '1999',
 '1999',
 '1997',
 '2001',
 '1998',
 '1999',
 '2000',
 '1998',
 '1998',
 '1998',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '2000',
 '2000',
 '1997',
 '1997',
 '1997',
 '1997',
 '2001',
 '1998',
 '1998',
 '1999',
 '1999',
 '1998',
 '1997',
 '1998',
 '1997',
 '2000',
 '2000',
 '2000',
 '1999',
 '1997',
 '1997',
 '2000',
 '1999',
 '1999',
 '1998',
 '1998',
 '1997',
 '1998',
 '1999',
 '1997',
 '1997',
 '1998',
 '1997',
 '2001',
 '1999',
 '1997',
 '1998',
 '1998',
 '1999',
 '1997',
 '2001',
 '2000',
 '2001',
 '1999',
 '1997',
 '2001',
 '1999',
 '1998',
 '1999',
 '2001',
 

In [61]:
#Cleaning Courses
#also sir you just said to clean the data which I will but I kind of want to shorten the names instead of making it the course title
#Also same technique done in Day 14
Unique_Dirty_Courses=set(Dirty_Courses)
Courses_Dict = {name:None for name in Unique_Dirty_Courses}

for i in Courses_Dict:
    cleaned_name = input(i+": "+"Enter name: ")
    Courses_Dict[i]=cleaned_name
Cleaned_Courses = [re.sub(r''.join(i),r''.join(Courses_Dict[i]),i) for i in Dirty_Courses]
Cleaned_Courses

Information Technology Entrepreneurship: Enter name: ITE
BS CTM: Enter name: CTM
Management Engineering: Enter name: ME
Management: Enter name: MGT
BS ME: Enter name: ME
CTM: Enter name: CTM
Communications Technology: Enter name: CTM
MGT: Enter name: MGT
BS LM: Enter name: LM
Legal Management: Enter name: LM
ME: Enter name: ME
LM: Enter name: LM
BS MGT: Enter name: MGT
ITE: Enter name: ITE
BS ITE: Enter name: ITE


['LM',
 'MGT',
 'CTM',
 'ITE',
 'CTM',
 'MGT',
 'LM',
 'ME',
 'ME',
 'CTM',
 'LM',
 'ME',
 'CTM',
 'MGT',
 'ITE',
 'LM',
 'ITE',
 'MGT',
 'ITE',
 'ME',
 'LM',
 'ITE',
 'MGT',
 'ME',
 'ME',
 'ME',
 'CTM',
 'LM',
 'LM',
 'CTM',
 'ITE',
 'MGT',
 'ITE',
 'LM',
 'ITE',
 'MGT',
 'MGT',
 'LM',
 'ME',
 'LM',
 'MGT',
 'LM',
 'MGT',
 'ITE',
 'ME',
 'ME',
 'MGT',
 'ITE',
 'LM',
 'LM',
 'ITE',
 'CTM',
 'MGT',
 'CTM',
 'ME',
 'ME',
 'CTM',
 'ME',
 'ITE',
 'ITE',
 'CTM',
 'ITE',
 'ME',
 'ITE',
 'ME',
 'LM',
 'CTM',
 'MGT',
 'CTM',
 'LM',
 'CTM',
 'MGT',
 'ME',
 'LM',
 'ITE',
 'CTM',
 'CTM',
 'MGT',
 'LM',
 'CTM',
 'CTM',
 'CTM',
 'ME',
 'ITE',
 'LM',
 'MGT',
 'ITE',
 'ME',
 'ITE',
 'ITE',
 'ITE',
 'ME',
 'ITE',
 'LM',
 'CTM',
 'CTM',
 'MGT',
 'ITE',
 'CTM',
 'MGT',
 'CTM',
 'LM',
 'ME',
 'CTM',
 'ITE',
 'CTM',
 'MGT',
 'MGT',
 'ME',
 'ME',
 'ME',
 'MGT',
 'ITE',
 'CTM',
 'ITE',
 'ITE',
 'LM',
 'ME',
 'LM',
 'LM',
 'LM',
 'LM',
 'LM',
 'MGT',
 'LM',
 'LM',
 'LM',
 'LM',
 'ME',
 'MGT',
 'CTM',
 'LM',


In [41]:
#Cleaning phonenumbers
#Similar method to years except using regex to find the 9
Cleaned_Phonenumbers=[]
for i in Dirty_Phonenumbers:
    Cleaned_Phonenumber = re.search(r'9.+', i).group()
    Cleaned_Phonenumbers.append("0"+Cleaned_Phonenumber)
Cleaned_Phonenumbers

['0918 302 6210',
 '0927 212 6276',
 '0918 253 7017',
 '0918 714 4360',
 '0927 462 1674',
 '0918 629 5268',
 '0917 369 7388',
 '0917 240 6000',
 '0918 226 5748',
 '0918 342 1717',
 '0927 708 1308',
 '0927 747 5497',
 '0917 959 4490',
 '0927 543 4192',
 '0917 211 9497',
 '0918 185 5162',
 '0927 146 8146',
 '0927 516 4280',
 '0927 958 6909',
 '0917 514 9105',
 '0927 984 7499',
 '0927 699 3694',
 '0917 317 2478',
 '0917 159 3829',
 '0918 898 6972',
 '0917 796 4622',
 '0927 138 7580',
 '0917 195 9244',
 '0917 643 3996',
 '0917 758 3119',
 '0918 669 4265',
 '0917 750 8539',
 '0927 236 9504',
 '0917 438 4447',
 '0917 773 6035',
 '0918 455 4185',
 '0927 889 9252',
 '0927 874 8083',
 '0927 842 3053',
 '0927 923 1234',
 '0918 381 4956',
 '0918 602 9278',
 '0918 532 6748',
 '0917 654 1291',
 '0918 244 1683',
 '0917 818 1190',
 '0917 468 6425',
 '0918 635 1688',
 '0917 821 1896',
 '0927 924 9506',
 '0918 854 2626',
 '0918 438 9745',
 '0917 893 8235',
 '0918 216 5250',
 '0917 257 1057',
 '0927 949

In [47]:
#Have to combine all the cleaned and old used data together
Cleaned_Data=zip(Firstnames,Lastnames,Cleaned_Birthmonths,Birthdays,Cleaned_Birthyears,Cleaned_Courses,Cleaned_Phonenumbers)
Cleaned_Data=list(Cleaned_Data)
Cleaned_Data
columns = ["Firstnames","Lastnames","Cleaned_Birthmonths","Birthdays","Cleaned_Birthyears","Cleaned_Courses","Cleaned_Phonenumbers"]
with open('cleanrecweekdata.csv', 'w') as csvFile:
    writer = csv.writer(csvFile)
    writer.writerow(columns)
    for i in Cleaned_Data:
            writer.writerow(i)

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [69]:
print(Birthdays.count('1'))
print(Birthdays.count('2'))
print(Birthdays.count('3'))
print(Birthdays.count('4'))
print(Birthdays.count('5'))
print(Birthdays.count('6'))
print(Birthdays.count('7'))
print(Birthdays.count('8'))
print(Birthdays.count('9'))
print(Birthdays.count('10'))
print(Birthdays.count('11'))
print(Birthdays.count('12'))
print(Birthdays.count('13'))
print(Birthdays.count('14'))
print(Birthdays.count('15'))
print(Birthdays.count('16'))
print(Birthdays.count('17'))
print(Birthdays.count('18'))
print(Birthdays.count('19'))
print(Birthdays.count('20'))
print(Birthdays.count('21'))
print(Birthdays.count('22'))
print(Birthdays.count('23'))
print(Birthdays.count('24'))
print(Birthdays.count('25'))
print(Birthdays.count('26'))
print(Birthdays.count('27'))
print(Birthdays.count('28'))
print(Birthdays.count('29'))
print(Birthdays.count('30'))
print(Birthdays.count('31'))


112
106
90
110
100
99
88
111
98
111
105
98
119
102
121
115
116
107
113
91
123
103
116
108
116
116
102
104
0
0
0


In [68]:

print(Cleaned_Courses.count('ME'))
print(Cleaned_Courses.count('ITE'))
print(Cleaned_Courses.count('CTM'))
print(Cleaned_Courses.count('MGT'))
print(Cleaned_Courses.count('LM'))

607
590
589
629
585


In [179]:
print(Cleaned_Birthmonths.count('1'))
print(Cleaned_Birthmonths.count('2'))
print(Cleaned_Birthmonths.count('3'))
print(Cleaned_Birthmonths.count('4'))
print(Cleaned_Birthmonths.count('5'))
print(Cleaned_Birthmonths.count('6'))
print(Cleaned_Birthmonths.count('7'))
print(Cleaned_Birthmonths.count('8'))
print(Cleaned_Birthmonths.count('9'))
print(Cleaned_Birthmonths.count('10'))
print(Cleaned_Birthmonths.count('11'))
print(Cleaned_Birthmonths.count('12'))

270
232
246
267
273
259
249
241
232
178
277
276


In [178]:
Globe=0
Smart=0
for number in Cleaned_Phonenumbers:
    if re.match(r'^0917.+.',number) or re.match(r'^0927.+.',number):
        Globe+=1
    else:
        Smart+=1
            
print(Globe) 
print(Smart)

2045
955


In [173]:
re.match(r'^0917','0911357')

In [131]:
print(Cleaned_Birthyears.count('2000'))
print(Cleaned_Birthyears.count('2001'))
print(Cleaned_Birthyears.count('1999'))
print(Cleaned_Birthyears.count('1998'))
print(Cleaned_Birthyears.count('1997'))

380
362
756
746
756
